WordPress/src/main/java/org/wordpress/android/util/DialogUtils.java

```diff
@@ -11,22 +14,55 @@
 import org.wordpress.android.widgets.OpenSansEditText;
 import org.wordpress.android.widgets.WPTextView;
 
-public class DialogUtils {
-    public static void showMyProfileDialog(Context context,
-                                           String title,
-                                           String initialText,
-                                           String hint,
-                                           boolean isMultiline,
-                                           final Callback callback) {
-        LayoutInflater layoutInflater = LayoutInflater.from(context);
+public class DialogUtils extends DialogFragment {
+
+    private static final String TITLE_TAG = "TITLE";
+    private static final String INITIAL_TEXT_TAG = "INITIAL-TEXT";
+    private static final String HINT_TAG = "HINT";
+    private static final String IS_MULTILINE_TAG = "IS-MULTILINE";
+
+    private static Context mContext;
+    private static Callback mCallback;
```

Comment:
Hey @rishabh7m , thanks for your effort!
So, the pattern is to actually cast it to the `interface` you're interested in, the `Callback` in this case and ofc have the parent activity implement it. To be on the safe side, you can first test that the parent activity is indeed an `instanceof` the interface before casting it. Let me know if this clears it up. Cheers!